src/tm_devices/driver_mixins/abstract_device_functionality/screen_capture_mixin.py

```diff
@@ -0,0 +1,99 @@
+"""A mixin class providing common methods for devices that can perform screen captures."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import final, Optional, Tuple, TYPE_CHECKING, Union
+
+if TYPE_CHECKING:
+    import os
+
+
+class ScreenCaptureMixin(ABC):
+    """A mixin class providing common methods for devices that can perform screen captures."""
+
+    @property
+    @abstractmethod
+    def valid_image_extensions(self) -> Tuple[str, ...]:
+        """Return a tuple of valid image extensions for this device.
+
+        The extensions will be in the format '.ext', where 'ext' is the lowercase extension,
+        e.g. (".png", ".jpg").
+
+        Returns:
+            Tuple[str, ...]: A tuple of valid, lowercase image extensions for this device.
+        """
+
+    @final
+    def save_screenshot(
+        self,
+        filename: Union[str, os.PathLike[str]],
+        *,
+        colors: Optional[str] = None,
+        view_type: Optional[str] = None,
+        local_folder: Union[str, os.PathLike[str]] = "./",
+        device_folder: Union[str, os.PathLike[str]] = "./",
+        keep_device_file: bool = False,
+    ) -> None:
+        """Capture a screenshot from the device and save it locally.
+
+        Args:
+            filename: The name of the file to save the screenshot as.
+            colors: The color scheme to use for the screenshot.
+            view_type: The type of view to capture.
```

Comment:
I don't understand this arg description, and I don't see it used in the tests or examples. I think it needs more context. and if it's something specific to scopes, let's discuss how to handle that.

Reply:
I updated the docstrings